src/v/cluster/log_eviction_stm.cc

```diff
@@ -40,82 +41,123 @@ log_eviction_stm::log_eviction_stm(
   ss::abort_source& as,
   storage::kvstore& kvstore)
   : persisted_stm("log_eviction_stm.snapshot", logger, raft, kvstore)
-  , _logger(logger)
-  , _as(as) {}
+  , _as(as)
+  , _queue(max_event_queue_size) {}
 
 ss::future<> log_eviction_stm::start() {
     ssx::spawn_with_gate(_gate, [this] { return monitor_log_eviction(); });
     ssx::spawn_with_gate(
-      _gate, [this] { return write_raft_snapshots_in_background(); });
+      _gate, [this] { return handle_log_eviction_events(); });
     return persisted_stm::start();
 }
 
 ss::future<> log_eviction_stm::stop() {
-    _reap_condition.broken();
+    _queue_mutex.broken();
+    _queue.abort(std::make_exception_ptr(ss::abort_requested_exception()));
     co_await persisted_stm::stop();
 }
 
-ss::future<> log_eviction_stm::write_raft_snapshots_in_background() {
+ss::future<> log_eviction_stm::enqueue_eviction_event(
+  model::offset offset, retry_event_until_success wait) {
+    vlog(
+      _log.trace,
+      "log eviction event at offset: {}, waiting for truncation: {}",
+      offset,
+      wait);
+    co_await _queue_mutex.get_units();
```

Comment:
I think theres a potential issue here, we should be storing the units returned by `get_units`

Reply:
in a local variable yes, :facepalm

Reply:
in a local variable yes, :facepalm

Reply:
yes, in a local variable :face_palm